### PR TITLE
Set up associations

### PR DESCRIPTION
**Pull request by shaqayq**
In this part of the project, I added associations between classes. I have created `Book `, `Rental `(intermediate class between `Person `and `Book `  classes), `Classroom `classes .

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  attr_accessor :title, :author, :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rental = []
+  end
+
+  def add_rental(date, _rental)
+    Rental.new(date, self, person)
```
Comment (ShadyShawkat) [inline book.rb]: - [ ] Kindly make sure to pass the right parameters to the `add_rental` method. Currently, you're not passing the `person` parameter to the `add_rental` method but you're passing it to the new rental. Kindly fix this.
- [ ] Please be mindful of the order of parameters passed in the new rental. The rental's constructor has the following paramters passed in it:
```
def initialize(date, person, book)
```
- Please follow the same order of parameters passed.

File: person.rb [hunk 2]
```diff
@@ -27,4 +28,8 @@ def correct_name
   def of_age?
     @age >= 18
   end
+
+  def add_rental(rental)
+    @rental << rental
+  end
```
Comment (ShadyShawkat) [inline person.rb]: - [ ] Kindly follow the same implementation of creating a rental like what you did in the `Book` class.

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi @shaqayq, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- All linters are passing ✔️ 
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 2 above)
Comment (shaqayq) [reply person.rb]: @ShadyShawkat !
Thank you for reviewing my code I will change it.

**Review by Reem-lab — APPROVED** [review]
# Hi @shaqayq,👋🏻👋🏻

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉🔥🔥💯💯





## Highlights ✨✨
📌 No Linters error ✅
📌 Descriptive pull request✅
📌 Great working on the functionality✅
📌 Good commit messages ✅

## Optional suggestions


_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**